lib/ibmiconnector.js

```diff
@@ -786,66 +899,103 @@ class IBMiConnector extends SQLConnector {
+   * statement in a `SELECT * FROM FINAL TABLE` to get the inserted id.
+   * @param {String} model The model name
+   * @param {Object} data The model data object
+   * @param {Object} options The options object
+   * @returns {string} The INSERT SQL statement
+   */
+  buildInsert(model, data, options) {
+    const idCol = this.idColumn(model);
+    if (typeof data[idCol] !== undefined && data[idCol] === null) {
+      data[idCol] = undefined;
+    }
+    const fields = this.buildFields(model, data);
+    const insertStmt = this.buildInsertInto(model, fields, options);
+    const columnValues = fields.columnValues;
+    const fieldNames = fields.names;
+    if (fieldNames.length) {
+      const values = ParameterizedSQL.join(columnValues, ',');
+      values.sql = 'VALUES(' + values.sql + ')';
+      insertStmt.merge(values);
+    } else {
+      insertStmt.merge(this.buildInsertDefaultValues(model, data, options));
+    }
+    const returning = this.buildInsertReturning(model, data, options);
+    if (returning) {
+      insertStmt.merge(returning);
+    }
+    const parameterizedStatement = this.parameterize(insertStmt);
+    parameterizedStatement.sql =
+      `SELECT * FROM FINAL TABLE(${parameterizedStatement.sql})`;
+    return parameterizedStatement;
+  }
+
+  buildInsertDefaultValues(model, data, options) {
+    const idColumnEscaped = this.idColumnEscaped(model);
+    return `(${idColumnEscaped}) VALUES (DEFAULT)`;
+  }
+
   /**
    * Parse the result for SQL INSERT for newly inserted id
    * @param {String} model Model name
    * @param {Object} info The status object from driver
    * @returns {*} The inserted id value
    */
   getInsertedId(model, info) {
-    const insertedId = info && typeof info.insertId === 'number' ? info.insertId : undefined;
-    return insertedId;
+    const idColumn = this.idColumn(model);
+    return info && info.length === 1 ? info[0][idColumn] : undefined;
   }
 
   /**
    * Execute a SQL statement with given parameters
-   * @param {String} sql The SQL statement
+   * @param {string} sql The SQL statement
    * @param {*[]} [params] An array of parameter values
    * @param {Object} [options] Options object
    * @param {Function} [callback] The callback function
    */
-  executeSQL(sql, params, options, callback) {
-    console.log(sql);
-    this.pool.connect((error1, conn) => {
-      if (error1) {
-        callback(error1);
-        return;
-      }
-
-      // Transaction
-      if (options.transaction) {
-        conn.beginTransaction((error2) => {
-          if (error2) {
-            callback(error2);
-            return;
-          }
-          conn.query(sql, params, (error3, data) => {
-            // conn.close();
-            return callback(error3, data);
-          });
+  async executeSQL(sql, params, options, callback) {
+    const transaction = options.transaction;
+    // running in a transaction
+    if (transaction && transaction.connector === this) {
```

Comment:
What if `transaction.connector` is not `this`? Is it a valid situation? If not, then we should let the user know - either by passing an error to the callback arg, or at least printing a warning to the console. 

Valid situation or not, I think it would be great to add a [`debug`](https://www.npmjs.com/package/debug) statement for that situation.